tests/test_perceptual_loss.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import unittest
+
+import torch
+from parameterized import parameterized
+
+from monai.losses import PerceptualLoss
+
+TEST_CASES = [
+    [{"spatial_dims": 2, "network_type": "squeeze"}, (2, 1, 64, 64), (2, 1, 64, 64)],
+    [
+        {"spatial_dims": 3, "network_type": "squeeze", "is_fake_3d": True, "fake_3d_ratio": 0.1},
+        (2, 1, 64, 64, 64),
+        (2, 1, 64, 64, 64),
+    ],
+    [{"spatial_dims": 2, "network_type": "radimagenet_resnet50"}, (2, 1, 64, 64), (2, 1, 64, 64)],
+    [{"spatial_dims": 2, "network_type": "radimagenet_resnet50"}, (2, 3, 64, 64), (2, 3, 64, 64)],
+    [
+        {"spatial_dims": 3, "network_type": "radimagenet_resnet50", "is_fake_3d": True, "fake_3d_ratio": 0.1},
+        (2, 1, 64, 64, 64),
+        (2, 1, 64, 64, 64),
+    ],
+    [
+        {"spatial_dims": 3, "network_type": "medicalnet_resnet10_23datasets", "is_fake_3d": False},
+        (2, 1, 64, 64, 64),
+        (2, 1, 64, 64, 64),
+    ],
+    [
+        {"spatial_dims": 3, "network_type": "resnet50", "is_fake_3d": True, "pretrained": True, "fake_3d_ratio": 0.2},
+        (2, 1, 64, 64, 64),
+        (2, 1, 64, 64, 64),
+    ],
+]
+
+
+class TestPerceptualLoss(unittest.TestCase):
```

Comment:
some issues with earlier versions  of torchvision
```
[2023-07-31T16:47:05.254Z] ======================================================================
[2023-07-31T16:47:05.254Z] ERROR: test_identical_input_6 (tests.test_perceptual_loss.TestPerceptualLoss)
[2023-07-31T16:47:05.254Z] ----------------------------------------------------------------------
[2023-07-31T16:47:05.254Z] Traceback (most recent call last):
[2023-07-31T16:47:05.254Z]   File "/usr/local/lib/python3.8/dist-packages/parameterized/parameterized.py", line 620, in standalone_func
[2023-07-31T16:47:05.254Z]     return func(*(a + p.args), **p.kwargs, **kw)
[2023-07-31T16:47:05.254Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_perceptual_loss.py", line 57, in test_identical_input
[2023-07-31T16:47:05.254Z]     loss = PerceptualLoss(**input_param)
[2023-07-31T16:47:05.254Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/losses/perceptual.py", line 86, in __init__
[2023-07-31T16:47:05.254Z]     self.perceptual_function = TorchvisionModelPerceptualSimilarity(
[2023-07-31T16:47:05.254Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/losses/perceptual.py", line 306, in __init__
[2023-07-31T16:47:05.254Z]     weights=torchvision.models.ResNet50_Weights.DEFAULT if pretrained else None
[2023-07-31T16:47:05.254Z] AttributeError: module 'torchvision.models' has no attribute 'ResNet50_Weights'
[2023-07-31T16:47:05.254Z] 
[2023-07-31T16:47:05.254Z] ======================================================================
[2023-07-31T16:47:05.254Z] ERROR: test_shape_6 (tests.test_perceptual_loss.TestPerceptualLoss)
[2023-07-31T16:47:05.254Z] ----------------------------------------------------------------------
[2023-07-31T16:47:05.254Z] Traceback (most recent call last):
[2023-07-31T16:47:05.254Z]   File "/usr/local/lib/python3.8/dist-packages/parameterized/parameterized.py", line 620, in standalone_func
[2023-07-31T16:47:05.254Z]     return func(*(a + p.args), **p.kwargs, **kw)
[2023-07-31T16:47:05.254Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/tests/test_perceptual_loss.py", line 51, in test_shape
[2023-07-31T16:47:05.254Z]     loss = PerceptualLoss(**input_param)
[2023-07-31T16:47:05.254Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/losses/perceptual.py", line 86, in __init__
[2023-07-31T16:47:05.254Z]     self.perceptual_function = TorchvisionModelPerceptualSimilarity(
[2023-07-31T16:47:05.254Z]   File "/home/jenkins/agent/workspace/MONAI-premerge/monai/monai/losses/perceptual.py", line 306, in __init__
[2023-07-31T16:47:05.254Z]     weights=torchvision.models.ResNet50_Weights.DEFAULT if pretrained else None
[2023-07-31T16:47:05.254Z] AttributeError: module 'torchvision.models' has no attribute 'ResNet50_Weights'
[2023-07-31T16:47:05.254Z] 
[2023-07-31T16:47:05.254Z] ----------------------------------------------------------------------
[2023-07-31T16:47:05.254Z] Ran 12501 tests in 757.933s
[2023-07-31T16:47:05.254Z] 
[2023-07-31T16:47:05.254Z] FAILED (errors=2, skipped=1190)
```

I think you can add decorators to skip the cases -- example:

https://github.com/Project-MONAI/MONAI/blob/4c22a27aa5bc0b2d9c41f6d9a3201cc8d99a53ed/tests/test_retinanet.py#L100-L101